pkg/driver/driver_aws.go

```diff
@@ -97,25 +97,32 @@ func (d *AWSDriver) Create() (string, string, error) {
 		return "Error", "Error", err
 	}
 
-	securityGroupIDs, err := d.generateSecurityGroups(d.AWSMachineClass.Spec.NetworkInterfaces[0].SecurityGroupIDs)
-	if err != nil {
-		return "Error", "Error", err
+	var networkInterfaceSpecs []*ec2.InstanceNetworkInterfaceSpecification
+	for i, netIf := range d.AWSMachineClass.Spec.NetworkInterfaces {
+		spec := &ec2.InstanceNetworkInterfaceSpecification{
+			Groups:                   aws.StringSlice(netIf.SecurityGroupIDs),
+			DeviceIndex:              aws.Int64(int64(i)),
+			AssociatePublicIpAddress: &netIf.AssociatePublicIPAddress,
```

Comment:
Can we please explicitly set `AssociatePublicIpAddress` to false if not specified in the MachineClass ? Unintentionally creating a public-address could have severe security-impact.
- I see, we are using `bool` type, please feel free to convert it to `*bool`. You can then check if it's `nil` (implies not set in MachineClass), simply set it to `false`. 
- This would also help us to not introduce the breaking API-change for Gardener, as it consumes MCM out of box.

Reply:
>I see, we are using bool type, please feel free to convert it to *bool. You can then check if it's nil (implies not set in MachineClass), simply set it to false.
I've converted the MachineClass' type, but I strongly believe that there is no need for `if type != nil` idiom in this case. You see, if we pass these parameters as-is, then it would exactly the same as not passing them at all since Go's pointer's empty value is `nil`.

Reply:
Sure, thanks.